test/TestFilePathEvaluation.py

```diff
@@ -0,0 +1,128 @@
+"""Testing file path evaluation when using import_tasks / include_tasks."""
+import os
+import shutil
+import tempfile
+import textwrap
+
+import pytest
+
+from ansiblelint.runner import Runner
+
+LAYOUT_IMPORTS = {
+    'directories': ['tasks', 'tasks/subtasks'],
+    'files': {
+        'main.yml': textwrap.dedent(
+            """\
+            ---
+            - hosts: target
+              gather_facts: false
+              tasks:
+                - name: from main import task 1
+                  import_tasks: tasks/task_1.yml
+            """
+        ),
+        'tasks/task_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 1 import task 2
+              import_tasks: tasks/task_2.yml
+            """
+        ),
+        'tasks/task_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 2 import subtask 1
+              import_tasks: tasks/subtasks/subtask_1.yml
+            """
+        ),
+        'tasks/subtasks/subtask_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 1 import subtask 2
+              import_tasks: tasks/subtasks/subtask_2.yml
+            """
+        ),
+        'tasks/subtasks/subtask_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 2 do something
+              debug:
+                msg: |
+                  Something...
+            """
+        ),
+    },
+}
+
+LAYOUT_INCLUDES = {
+    'directories': ['tasks', 'tasks/subtasks'],
+    'files': {
+        'main.yml': textwrap.dedent(
+            """\
+            ---
+            - hosts: target
+              gather_facts: false
+              tasks:
+                - name: from main import task 1
+                  include_tasks: tasks/task_1.yml
+            """
+        ),
+        'tasks/task_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 1 import task 2
+              include_tasks: tasks/task_2.yml
+            """
+        ),
+        'tasks/task_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 2 import subtask 1
+              include_tasks: tasks/subtasks/subtask_1.yml
+            """
+        ),
+        'tasks/subtasks/subtask_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 1 import subtask 2
+              include_tasks: tasks/subtasks/subtask_2.yml
+            """
+        ),
+        'tasks/subtasks/subtask_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 2 do something
+              debug:
+                msg: |
+                  Something...
+            """
+        ),
+    },
+}
+
+
```

Comment:
:point_down: These XPASS entries mean that the test passes under those envs while it's expected to always fail until the code is fixed. This means that the test may be unstable and does not unambiguously catch the bug.